### DISCUSSION TITLE
How to generate from a skeleton branch?

### DISCUSSION THREAD

**Discussion by elpiel** [cargo-generate/cargo-generate#1207] [Q&A] [answered]
**Describe the feature**

There are certain situations, like making a workshop, where you would like to have a `skeleton` branch with all of the code removed for the workshop to use with cargo-generate but another `development` or `main` branch for the workshop application code in a full and working state.

**Comment by sassman** [comment, marked as answer]
Curious, and whats the part stopping you creating those branches on a repo? And use the `--branch` argument of cargo generate. 

Like `cargo generate --branch skeleton gh:your/repo`

Or rephrases whats the missing feature exactly, maybe you can elaborate?

**Comment by elpiel** [comment]
yep, I didn't find it in the book and I didn't check the cli usage :facepalm: 
Thanks!

**Comment by sassman** [comment]
No worries, maybe this example is a good case to illustrate in the docs. It's not that unlikely other may encounter that question too.
